video/out/wayland_common.c

```diff
@@ -988,21 +995,20 @@ static void surface_handle_preferred_buffer_scale(void *data,
                                                   int32_t scale)
 {
     struct vo_wayland_state *wl = data;
-    double old_scale = wl->scaling;
 
-    if (wl->fractional_scale_manager)
+    if (wl->fractional_scale_manager || wl->scaling == scale)
         return;
 
-    wl->scaling = scale;
+    wl->pending_scaling = scale;
     wl->scale_configured = true;
     MP_VERBOSE(wl, "Obtained preferred scale, %f, from the compositor.\n",
                wl->scaling);
     wl->pending_vo_events |= VO_EVENT_DPI;
-    if (wl->current_output) {
-        rescale_geometry(wl, old_scale);
-        set_geometry(wl, false);
-        prepare_resize(wl);
-    }
+    wl->need_rescale = true;
+
+    // Update scaling now.
+    if (single_output_spanned(wl))
+        update_output_scaling(wl);
```

Comment:
There's no advantage in doing that. All you do is change which part of the window is affected by compositor-side scaling.

Reply:
>conveys this information with the scale events to make sure that a larger portion of the window is displayed at the correct scale

I don't think anybody does this in practice. wlroots sends `preferred_scale` as soon as the window touches a new output so with your proposal what will happen is that the window will rescale if it 1% touches the new one while it is 99% on the old one.

Reply:
Alright but not everyone does so it's not like there's a universal advantage. I don't think trying to optimize the overlap case is worthwhile. Your picture is messed up anyway.

Reply:
>my purposed change does no harm either

But it does. On sway, a 1% overlap on the new monitor will immediately scale despite the most of the window being over the old monitor (which is what master currently does of course). If hypothetically I move the monitor partially into the new output for a bit but then move it back, I degrade scaling for most of the window.

If we got absolute coordinates or such, it could be worth optimizing but there's no way to predict what a compositor will do. You'll regress something either way.